_SPIKE_.md

```diff
@@ -0,0 +1,178 @@
+# Spike documentation
+
+See the discussion in
+[#2082](https://github.com/strongloop/loopback-next/issues/2082) and linked
+issue for the discussions we already have had in the past.
+
+The code in this spike is building on top of the spike for Inclusion of related
+models ([PR#2592)(https://github.com/strongloop/loopback-next/pull/2592)),
+please ignore the first commit.
+
+## Overview
+
+I am proposing to leverage the interface `JsonSchemaOptions` and
+`getModelSchemaRef` helper to build schema with additional constraints. Later
+on, we can explore different ways how to enable includeRelations flag via
+OpenAPI spec extensions.
+
+### Exclude properties from CREATE requests
+
+An example showing a controller method excluding the property `id`:
+
+```ts
+class TodoListController {
+  // ...
+
+  @post('/todo-lists', {
+    responses: {
+      // left out for brevity
+    },
+  })
+  async create(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(TodoList, {exclude: ['id']}),
+          /***** ^^^ THIS IS IMPORTANT - OPENAPI SCHEMA ^^^ ****/
+        },
+      },
+    })
+    obj: Pick<TodoList, Exclude<keyof TodoList, 'id'>>,
```

Comment:
Maybe define a type alias:
```ts
export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
```

Reply:
+1 to define a type alias.

We must carefully choose a name that will be different enough from `Exclude`. Few options that come to my mind:
- `ExcludeProperties<T, K>`
- `ExcludeKeys<T, K>`
- `OmitProperties<T, K>`
- `OmitKeys<T, K>`